docs/internals.rst

```diff
@@ -185,3 +185,138 @@ Managers
 The Manager acts as a gatekeeper for the request/response lifecycle.  It is
 unlikely that you will need to change the Manager as most functionality can be
 implemented in the Middleware layer.
+
+
+Request Processing for Persistent Connection Providers
+------------------------------------------------------
+
+.. py:class:: web3.providers.websocket.request_processor.RequestProcessor
+
+The ``RequestProcessor`` class is responsible for the storing and syncing up of
+asynchronous requests to responses for a ``PersistentConnectionProvider``. The best
+example of one such provider is the
+:class:`~web3.providers.websocket.WebsocketProviderV2`. In order to send a websocket
+message and receive a response to that particular request,
+``PersistentConnectionProvider`` instances have to match request *id* values to
+response *id* values coming back from the websocket connection. Any provider that does
+not adhere to the `JSON-RPC 2.0 specification <https://www.jsonrpc.org/specification>`_
+in this way will not work with ``PersistentConnectionProvider`` instances. The specifics
+of how the request processor handles this is outlined below.
+
+One-To-One Requests
+~~~~~~~~~~~~~~~~~~~
+
+One-to-one requests can be summarized as any request that expects one response back.
+An example is using the ``eth`` module API to request the latest block number.
+
+.. code-block:: python
+
+    >>> async def wsV2_one_to_one_example():
+    ...     async with AsyncWeb3.persistent_websocket(
+    ...         WebsocketProviderV2(f"ws://127.0.0.1:8546")
+    ...     ) as w3:
+    ...         # make a request and expect a single response returned on the same line
+    ...         latest_block_num = await w3.eth.block_number
+
+    >>> asyncio.run(wsV2_one_to_one_example())
+
+With websockets we have to call ``ws_send()`` and asynchronously receive responses via
+``ws.recv()``. In order to make the one-to-one request-to-response call work, we
+have to save the request information somewhere so that, when the response is received,
+we can match it to the original request that was made (the request with a matching *id*
+to the response that was received), and use that request information to process the
+response. Processing the response, in this case, means running it through the
```

Comment:
Long sentence, tried to clean it up a little but no hard feelings about it.
```suggestion
``ws.recv()``. A one-to-one request-to-response call works by matching the request 
and response to ensure the result is processed appropriately. When a request is made,
a request ID is stored so that it can be used when a response is returned to determine 
if it came from that same request. When the original request is found it processes the 
result by running it through the
```